### PR TITLE
CLN: clean IntervalArray._simple_new

### PR DESCRIPTION
**Pull request by topper-123**
There is a lot of code in `IntervalArray._simple_new`. I suggest moving all the code there that checks that the input data is correct into its own method (`_ensure_simple_new_inputs`), so we have a better split between building the ÌntervalArray` and checking that its inputs are correct.

This PR changes no functionality.

### CODE REVIEW

File: pandas/core/arrays/interval.py [hunk 1]
```diff
@@ -269,23 +272,39 @@ def _simple_new(
         cls: type[IntervalArrayT],
         left,
         right,
-        closed: IntervalClosedType | None = None,
-        copy: bool = False,
-        dtype: Dtype | None = None,
+        dtype: IntervalDtype,
```
Comment (jbrockmendel) [inline pandas/core/arrays/interval.py:275]: can we type left/right?

File: pandas/core/arrays/interval.py [hunk 2]
```diff
@@ -269,23 +272,39 @@ def _simple_new(
...
         verify_integrity: bool = True,
```
Comment (jbrockmendel) [inline pandas/core/arrays/interval.py:276]: ideally we'd get verify_integrity out of _simple_new

File: pandas/core/arrays/interval.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/arrays/interval.py:281]: Not easily. There are lines here where there are comparison operations (<=, >= etc.) and `ExtensionArray` doesn't have those.

I guess we could implement a `ExtensionArraySupportsComparisons(ExtensionArray)` but IMO that's out of scope for this PR (and I don't know how much work would be in getting that to that).

File: pandas/core/arrays/interval.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/arrays/interval.py]: I've made an implementation for this, see latest commit.

**Comment by topper-123** [comment]
The arm failure looks unrelated.

**Comment by topper-123** [comment]
The ARM failure looks unrelated.

File: pandas/core/arrays/interval.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/interval.py:281]: in practice isnt it always `ndarray | DatetimeArray | TimedeltaArray`?

Comment (topper-123) [reply pandas/core/arrays/interval.py:281]: I'm not sure, probably in practice yes. but typing it as `ndarray | DatetimeArray | TimedeltaArray` would mean that users can't use arrays  inherited from `ExtensionArray` as left/right, which would seem overtly restrictive?

I can try to add `#type ignore[x]` in various places, see if that is an acceptable solution.

Comment (jbrockmendel) [reply pandas/core/arrays/interval.py:281]: > which would seem overtly restrictive?

i dont understand this.  ndarray | DatetimeArray | TimedeltaArray _is_ what is allowed.  changing that would be a lot more invasive

Comment (topper-123) [reply pandas/core/arrays/interval.py:281]: Ok, the docs say array-like, which I interpreted as `ndarray | ExtensionArray`. Should the docs be more explicit that what is allowed in `IntervalArray` is `ndarray | DatetimeArray | TimedeltaArray`?

Comment (jbrockmendel) [reply pandas/core/arrays/interval.py:281]: seems reasonable

**Comment by topper-123** [comment]
I've updated the PR.

File: pandas/core/arrays/interval.py [hunk 3]
```diff
@@ -102,6 +105,7 @@
 
 
 IntervalArrayT = TypeVar("IntervalArrayT", bound="IntervalArray")
+IntervalSideT = Union[TimeArrayLike, np.ndarray]
```
Comment (jbrockmendel) [inline pandas/core/arrays/interval.py:108]: nice.  an option if you want to be more specific can specify `npt.NDArray[numeric_dtypes]`.  OK to call this out of scope

**Review by jbrockmendel — APPROVED** [review]

File: pandas/core/arrays/interval.py (hunk 3 above)
Comment (topper-123) [reply pandas/core/arrays/interval.py:108]: I can do it in a follow-up? I think I'd prefer an iterative approach, i.e. closing the book on this method cleaning PR and picking up the typing issue as a new PR.

**Comment by jbrockmendel** [comment]
thanks @topper-123

**Pull request merged**